packages/rpc-graphql/src/resolvers/account.ts

```diff
@@ -1,141 +1,196 @@
 import { Address } from '@solana/addresses';
+import { Commitment, Slot } from '@solana/rpc-types';
 import { GraphQLResolveInfo } from 'graphql';
 
 import { RpcGraphQLContext } from '../context';
-import { AccountLoaderArgs } from '../loaders';
-import { onlyPresentFieldRequested } from './resolve-info';
-
-// eslint-disable-next-line @typescript-eslint/no-explicit-any
-function transformParsedAccountData(parsedAccountData: any) {
-    const {
-        parsed: { info: result, type: accountType },
-        program: programName,
-        programId,
-    } = parsedAccountData;
-    // Tells GraphQL which account type has been
-    // returned by the RPC.
-    result.accountType = accountType;
-    result.programId = programId;
-    // Tells GraphQL which program the returned
-    // account belongs to.
-    result.programName = programName;
-    return result;
+import { AccountLoaderValue } from '../loaders';
+import { buildAccountLoaderArgSetFromResolveInfo, onlyFieldsRequested } from './resolve-info';
+
+type Encoding = 'base58' | 'base64' | 'base64+zstd';
+type DataSlice = { length: number; offset: number };
+type EncodedData = {
+    data: string;
+    dataSlice: DataSlice | null;
+};
+
+export class EncodedAccountData {
+    base58: EncodedData[] = [];
+    base64: EncodedData[] = [];
+    'base64+zstd': EncodedData[] = [];
+
+    insert(data: string, dataSlice: DataSlice | null, encoding: Encoding) {
+        this[encoding].push({ data, dataSlice });
+    }
+
+    get(dataSlice: DataSlice | null, encoding: Encoding): string | undefined {
+        return this[encoding].find(entry => {
+            if (dataSlice === null) {
+                return entry.dataSlice === null;
+            }
+            return entry.dataSlice?.length === dataSlice.length && entry.dataSlice?.offset === dataSlice.offset;
```

Comment:
I feel like you could just do this with an object whose keys are `base58:0:10` (encoding/offset/length).

Reply:
Nice call, I implemented this!